File: javapatterns/src/main/java/org/iut/nantes/factories/banque/FactoryCarte.java
package org.iut.nantes.factories.banque;

interface FactoryCarte
{
  Carte creerCarte( String proprietaire );
}

class FactoryCarteVisa implements FactoryCarte
{
  public Carte creerCarte( String proprietaire )
  {
    return new Visa( proprietaire );
  }
}

class FactoryCarteMC implements FactoryCarte
{
  public Carte creerCarte( String proprietaire )
  {
    return new MasterCard( proprietaire );
  }
}

class FactoryCarteAE implements FactoryCarte
{
  public Carte creerCarte( String proprietaire )
  {
    return new AmericanExpress( proprietaire );
  }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/factories/banque/FactoryCarte.kt
package org.iut.nantes.factories.banque

interface FactoryCarte {
    fun creerCarte(proprietaire: String): Carte
}

class FactoryCarteVisa : FactoryCarte {
    override fun creerCarte(proprietaire: String): Carte {
        return Visa(proprietaire)
    }
}

class FactoryCarteMC : FactoryCarte {
    override fun creerCarte(proprietaire: String): Carte {
        return MasterCard(proprietaire)
    }
}

class FactoryCarteAE : FactoryCarte {
    override fun creerCarte(proprietaire: String): Carte {
        return AmericanExpress(proprietaire)
    }
}